src/StreamFactoryInterface.php

```diff
@@ -7,17 +7,32 @@
 interface StreamFactoryInterface
 {
     /**
-     * Create a new stream from a resource or a string.
+     * Create a new stream from a resource.
      *
-     * If the argument is a resource, it MUST be readable and SHOULD be seekable. It MAY be writable.
-     * If the argument is a string, a temporary resource will be created that is writable and seekable.
-     * 
-     * If the arugment is a string it will be interpided as the content of the stream. File names or 
-     * file paths are not supported. 
-     *
-     * @param string|resource $resource
+     * @param resource $resource
      *
      * @return StreamInterface
      */
     public function createStream($resource);
+
+    /**
+     * Create a new stream from a file.
+     *
+     * @param string $path
+     * @param string $mode
+     *
+     * @return StreamInterface
+     */
+    public function createStreamFromFile($path, $mode = 'r');
+
+    /**
+     * Create a new stream from a string.
+     *
+     * String content will be used as the body of the stream.
+     *
+     * @param string $content
+     *
+     * @return StreamInterface
+     */
+    public function createStreamFromString($content);
```

Comment:
I know the argument type is a string, but a string is not always a string.

For example, you might create a stream with an image:

    $body = $factory->createStreamFromString(file_get_contents("image.jpg"));

Semantically the method-name doesn't make much sense in this case.

Perhaps `createStreamWithContent` might be more semantic? I think it would still make sense in the case where the stream is being populated with text or HTML for example, but perhaps makes more sense in other cases where the content is binary?